src/connector/src/parser/mod.rs

```diff
@@ -834,8 +834,26 @@ async fn into_chunk_stream_inner<P: ByteStreamSourceParser>(
                     }
                 },
 
-                Ok(ParseResult::SchemaChange(_)) => {
-                    // TODO
+                Ok(ParseResult::SchemaChange(schema_change)) => {
+                    if schema_change.is_empty() {
+                        continue;
+                    }
+
+                    let (oneshot_tx, oneshot_rx) = tokio::sync::oneshot::channel();
+                    // we bubble up the schema change event to the source executor via channel,
+                    // and wait for the source executor to finish the schema change process before
+                    // parsing the following messages.
+                    if let Some(ref tx) = parser.source_ctx().schema_change_tx {
+                        tx.send((schema_change, oneshot_tx))
+                            .await
+                            .expect("send schema change to executor");
+                        match oneshot_rx.await {
+                            Ok(()) => {}
```

Comment:
The actor (executor, and connector stream) will be dropped and replaced once the schema change succeeds, so I suppose this line will never be executed?

Reply:
> The actor (executor, and connector stream) will be dropped and replaced once the schema change succeeds

I think not. The `source` is not included in the streaming graph of the Table to be replaced. e.g.
the source actor will not be replaced in below example.
```
cdc source -> stream cdc scan -> mview (Table 1)
          +-> stream cdc scan -> mview (Table 2)
```

Reply:
Ah, this is true. 😄